src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceLookup/CallSiteFactoryTest.cs

```diff
@@ -6,16 +6,51 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using System.Reflection;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.DotNet.RemoteExecutor;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Microsoft.Extensions.DependencyInjection.Tests;
 using Xunit;
 
 namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     public class CallSiteFactoryTest
     {
+        [Fact]
+        public void GetService_FactoryCallSite_Transient_DoesNotFail()
+        {
+            IServiceProvider serviceProvider = null;
```

Comment:
If you declare this as `ServiceProvider`, you can make it:

```C#
using ServiceProvider serviceProvider = collection..BuildServiceProvider(ServiceProviderMode.Dynamic);
```

and get rid of the try-finally